cli/module_generate/scripts/generate_stubs.py

```diff
@@ -79,26 +145,14 @@ def update_annotation(annotation):
             elif isinstance(stmt, ast.ClassDef) and stmt.name == resource_name:
                 for cstmt in stmt.body:
                     if isinstance(cstmt, ast.ClassDef):
-                        nodes.append(cstmt.name)
+                        subclasses.append(return_subclass(resource_name, cstmt))
                     elif isinstance(cstmt, ast.AnnAssign):
-                        nodes.append(cstmt.target.id)
+                        nodes.add(cstmt.target.id)
                     elif isinstance(cstmt, ast.AsyncFunctionDef):
-                        for arg in cstmt.args.args:
-                            arg.annotation = update_annotation(arg.annotation)
-
-                        cstmt.body = [
-                            ast.Raise(
-                                exc=ast.Call(
-                                    func=ast.Name(id='NotImplementedError', ctx=ast.Load()),
-                                    args=[],
-                                    keywords=[]),
-                                cause=None,
-                                )
-                        ]
-                        cstmt.decorator_list = []
-                        if isinstance(cstmt.returns, ast.Name) and cstmt.returns.id in nodes:
-                            cstmt.returns = return_attribute(resource_name, cstmt.returns.id)
-                        indented_code = '\n'.join(['    ' + line for line in ast.unparse(cstmt).splitlines()])
+                        replace_async_func(resource_name, cstmt, nodes)
+                        indented_code = '\n'.join(
+                            ['    ' + line for line in ast.unparse(cstmt).splitlines()]
+                        )
```

Comment:
is it okay that we don't have an else statement at the end to catch any cases that do not fit any of the previous if or elif conditions?

Reply:
We do not need the others.